### DISCUSSION TITLE
9.2.1 🚀

### DISCUSSION THREAD

**Discussion by preardon** [BrighterCommand/Brighter#2304] [General]
## Changes

- Examples should show Darker attributes too @iancooper (#2300)
- [Snyk] Security upgrade rabbitmq from management to 3-management @snyk-bot (#2254)

## 🚀 Features

- Initial Plumb in of Bulk Deposit Async @preardon (#2264)
- Add initial OpenTelemetry (Tracing) Support @preardon (#2285)

<hr /><em>This discussion was created from the release <a href='https://github.com/BrighterCommand/Brighter/releases/tag/9.2.1'>9.2.1 🚀</a>.</em>

**Comment by wozzo** [comment]
This version seems to add a breaking change not identified here.
[Comparing 9.1.20 to 9.2.1](https://github.com/BrighterCommand/Brighter/compare/9.1.20...9.2.1#diff-3cd954e3f47cc1e74d68a593f875a187a3e9b726af4692dd014a9ba26f6a7357) there is a new property added to the `IRequest` interface.
We're currently using the `ICommand` and `IEvent` interfaces in our code.
Should we instead be using the `Command` and `Event` classes instead?

**Comment by iancooper** [comment]
@wozzo Normally we would not consider the addition of a new field a breaking change, but you are right, this is because we would expect you to derive from Command and Event. We do expose most of our interfaces, to let you do your own thing, but in this case you would end up forced to implement this property if you derive your own types from this interface, so it becomes a breaking change for you.

Normally, we flag a change to an interface that is additive as a "minor" version bump, not a major version bump. Otherwise fixes tend to just increment the version.

**Reply by iancooper** [reply]
PS I am aware for some folks coming from other .NET libs, our use of base class over interface is not as expected. Part of this is we don't recommend using assemblies to share types between CI boundaries (i.e. microservices).

**Reply by wozzo** [reply]
Thanks for the feedback. [The documentation](https://brightercommand.gitbook.io/paramore-brighter-documentation/brighter-request-handlers-and-middleware-pipelines/implementingahandler) shows using the `IRequest` interface.
```csharp
public class GreetingCommand : IRequest
{
    public GreetingCommand(string name)
    {
        Id = Guid.NewGuid();
        Name = name;
    }

    public Guid Id { get; set; }
    public string Name { get; private set; }
}
```
Is this an opportunity to improve the documentation?

**Comment by iancooper** [comment]
@wozzo Good spot, we should. Happy to do it, but also happy for you to raise a PR if you are keen.

**Reply by wozzo** [reply]
I would, except I've been doing it wrong it seems, so prob not the best person to be writing the documentation 🤣
I still don't know much about what the `Activity Span` property does for instance and whether anything should be done with that or not?
I'm also a bit confused by what you meant when you said 👇 and feel understanding that would be useful in writing the documentation.
> Part of this is we don't recommend using assemblies to share types between CI boundaries (i.e. microservices)

**Reply by iancooper** [reply]
@wozzo No worries, I'll pick it up. And thanks for the feedback, it all goes to making it clearer for someone else when you ask!

**Reply by iancooper** [reply]
@wozzo See https://github.com/BrighterCommand/Docs/issues/32

**Comment by iancooper** [comment]
@wozzo New docs: https://brightercommand.gitbook.io/paramore-brighter-documentation/brighter-request-handlers-and-middleware-pipelines/requests-commands-and-events
